src/main/java/com/revature/controllers/ProductController.java

```diff
@@ -40,11 +41,24 @@ public ResponseEntity<Product> getProductById(@PathVariable("id") int id) {
     }
 
     @Authorized
-    @PutMapping
+    @AuthorizedAdmin
+    @PutMapping("/create-update")
     public ResponseEntity<Product> upsert(@RequestBody Product product) {
+    	Optional<Product> curr_pd = productService.findById(product.getId());
```

Comment:
@XinChenCSC Please rename `curr_pd` to `currPd` to comply with camel casing conventions for naming local variables.

Here is the link to the sonar cloud report of this code smell: 
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=4&id=Revature-Sophia-ECommerce-1376_e-commerce-backend&open=AYIj3ZtF1IANdIAAKhTC

Reply:
fixed it